### ISSUE TITLE
Training is too slow and even the gpu usage is low

### ISSUE THREAD

**Issue by poornimajd** [nianticlabs/monodepth2#157]
I am training on a RTX 2080 ti gpu but the gpu usage is not above 55%  and most of the times its less than 10%.My batch size is 12 and I have around  15604 images in training and 2k images in validation set.It shows Training is using : CUDA .
here is the screenshot of my training
![Screenshot from 2020-03-10 10-55-11](https://user-images.githubusercontent.com/31023599/76282618-358d9180-62be-11ea-8ec2-658bc0c19435.png)
I am loading data from a hdd
Can you please help me out as to what is going wrong?

**Comment by mrharicot** [comment]
> I am loading data from a hdd
There you go! Your data should be on an SSD to maximize GPU usage.

**Comment by poornimajd** [comment]
Thank you for such a quick reply!!
I would like to know apart from ssd is there any other workaround?

**Comment by mrharicot** [comment]
You could downsample the images to the network input size first, but I doubt it will help much, HDDs just struggle doing a lot of read operation per second.
```bash
find kitti_data/ -name '*.png' | parallel 'convert -resize 640x192 -quality 92 -sampling-factor 2x2,1x1,1x1 {.}.png {.}.jpg && rm {}'
```
You could also use lru cache on the image loading function to cache some input images (or all if you have a lot of memory).
https://docs.python.org/3/library/functools.html#functools.lru_cache

**Comment by poornimajd** [comment]
I will check out these methods and see if it works.
Thanks for the reply !

**Comment by mrharicot** [comment]
Your best bet will still be to invest in an SSD, no need to go for expensive nvme, a sata ssd should be fine for most cases.
